src/ethereum/constantinople/vm/interpreter.py

```diff
@@ -164,6 +166,51 @@ def process_create_message(message: Message, env: Environment) -> Evm:
     return evm
 
 
+def process_create2_message(message: Message, env: Environment) -> Evm:
+    """
+    Executes a call to create a smart contract via CREATE2 opcode.
+
+    Parameters
+    ----------
+    message :
+        Transaction specific items.
+    env :
+        External items required for EVM execution.
+
+    Returns
+    -------
+    evm: :py:class:`~ethereum.constantinople.vm.Evm`
+        Items containing execution specific objects.
+    """
+    # take snapshot of state before processing the message
+    begin_transaction(env.state)
+
+    # It's expected that the creation operation works on empty storage. Hence
+    # we delete the storage and restore the account's state if there is an
+    # error in the initialization code execution.
+    destroy_storage(env.state, message.current_target)
+
+    increment_nonce(env.state, message.current_target)
+    evm = process_message(message, env)
+    if not evm.has_erred:
+        contract_code = evm.output
+        contract_code_gas = len(contract_code) * GAS_CODE_DEPOSIT
+        try:
+            evm.gas_left = subtract_gas(evm.gas_left, contract_code_gas)
+            ensure(len(contract_code) <= MAX_CODE_SIZE, OutOfGasError)
+        except OutOfGasError:
```

Comment:
I think peter changed this exception to `ExceptionalHalt`

Reply:
`OutOfGasError` is now a subclass of `ExceptionalHalt`: https://github.com/ethereum/execution-specs/blob/6a6f43417c73154d814262fe7dfb0f3f355d2248/src/ethereum/constantinople/vm/error.py#L52